opentelemetry-sdk/src/metrics/internal/exponential_histogram.rs

```diff
@@ -333,135 +330,55 @@ where
         this.record(value);
     }
 
-    fn clone_and_reset(&self, init: &BucketConfig) -> Self {
+    fn clone_and_reset(&self, init: &ExpoHistogramBucketConfig) -> Self {
         let mut current = self.lock().unwrap_or_else(|err| err.into_inner());
         let cloned = replace(current.deref_mut(), ExpoHistogramDataPoint::new(init));
         Mutex::new(cloned)
     }
 }
 
 #[derive(Debug, Clone, Copy, PartialEq)]
-struct BucketConfig {
-    max_size: i32,
-    max_scale: i8,
+pub(crate) struct ExpoHistogramBucketConfig {
+    pub(crate) max_size: i32,
+    pub(crate) max_scale: i8,
 }
 
 /// An aggregator that summarizes a set of measurements as an exponential
 /// histogram.
 ///
 /// Each histogram is scoped by attributes and the aggregation cycle the
 /// measurements were made in.
-pub(crate) struct ExpoHistogram<T: Number> {
-    value_map: ValueMap<Mutex<ExpoHistogramDataPoint<T>>>,
-    init_time: AggregateTimeInitiator,
-    temporality: Temporality,
-    filter: AttributeSetFilter,
-    record_sum: bool,
-    record_min_max: bool,
+pub(crate) struct ExpoHistogram<AC> {
+    pub(crate) aggregate_collector: AC,
```

Comment:
It's better to use the new method. That way we cannot mistakenly update these fields outside of the struct creation.